pad.c

```diff
@@ -954,6 +954,14 @@ S_pad_check_dup(pTHX_ PADNAME *name, U32 flags, const HV *ourstash)
             --off;
         }
     }
+    /* check the package for my sub &x vs. sub PACKAGE::x collisions */
+    if (PadnamePV(name)[0] == '&') {
+        CV *cv = get_cvn_flags(PadnamePV(name)+1, PadnameLEN(name)-1, GV_NOTQUAL);
```

Comment:
padnames are always UTF8, this needs the `SVf_UTF8` flag:
```
tony@venus:.../git/perl6$ git log -n1 --pretty=oneline
5db4884c69252a594467022d9fd3fcc1b79af84b (HEAD -> 21915-lexical-shadow) perldelta for new lexical sub warning
tony@venus:.../git/perl6$ cat ../21915.pl
use utf8;
use strict;
use warnings;

sub π {}
sub pi {}
my sub π {}
my sub pi {}
tony@venus:.../git/perl6$ ./perl -Ilib ../21915.pl
Lexical subroutine &pi masks previously declared package subroutine at ../21915.pl line 8.
tony@venus:.../git/perl6$
```

Reply:
Fixed and forcepushed. Your example now warns on both for me.